internal/services/orbital/contact_profile_resource_test.go

```diff
@@ -0,0 +1,245 @@
+  links {
+    channels {
+      name                 = "channelname"
+      bandwidth_mhz        = 100
+      center_frequency_mhz = 101
+      end_point {
+        end_point_name = "AQUA_command"
+        ip_address     = "10.0.1.0"
+        port           = "49513"
+        protocol       = "TCP"
+      }
+    }
+    direction    = "Uplink"
+    name         = "RHCP_UL"
+    polarization = "RHCP"
+  }
+  network_configuration_subnet_id = azurerm_subnet.test.id
+}
+`, template, data.RandomInteger)
+}
+
+func (r ContactProfileResource) update(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%[1]s
+
+resource "azurerm_orbital_contact_profile" "test" {
+  name                              = "testcontactprofile-%[2]d"
+  resource_group_name               = azurerm_resource_group.test.name
+  location                          = azurerm_resource_group.test.location
+  minimum_variable_contact_duration = "PT2M"
+  auto_tracking                     = "disabled"
+  links {
+    channels {
+      name                 = "channelname"
+      bandwidth_mhz        = 102
+      center_frequency_mhz = 103
+      end_point {
+        end_point_name = "AQUA_command"
+        ip_address     = "10.0.1.0"
+        port           = "49515"
+        protocol       = "TCP"
+      }
+    }
+    direction    = "Downlink"
+    name         = "RHCP_UL"
+    polarization = "RHCP"
+  }
+  network_configuration_subnet_id = azurerm_subnet.test.id
+}
+`, template, data.RandomInteger)
+}
+
+func (r ContactProfileResource) complete(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%[1]s
+
+resource "azurerm_eventhub_namespace" "test" {
+  name                = "eventhubtest-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  sku                 = "Standard"
+  capacity            = 1
+
+  tags = {
+    environment = "Production"
+  }
+}
+
+resource "azurerm_eventhub" "test" {
+  name                = "testeventhub-%d"
+  namespace_name      = azurerm_eventhub_namespace.test.name
+  resource_group_name = azurerm_resource_group.test.name
+  partition_count     = 2
+  message_retention   = 1
+}
+
+resource "azurerm_orbital_contact_profile" "test" {
+  name                              = "testcontactprofile-%d"
+  resource_group_name               = azurerm_resource_group.test.name
+  location                          = azurerm_resource_group.test.location
+  minimum_variable_contact_duration = "PT1M"
+  auto_tracking                     = "disabled"
+  event_hub_uri                     = azurerm_eventhub.test.id
+  minimum_elevation_degrees         = 12.12
+  links {
+    channels {
+      name                       = "channelname"
+      bandwidth_mhz              = 100
+      center_frequency_mhz       = 101
+      demodulation_configuration = "na"
+      modulation_configuration   = "AQUA_UPLINK_BPSK"
```

Comment:
these seem to just take a constant? so
```suggestion
      demodulation_type = "na"
      modulation_type   = "AQUA_UPLINK_BPSK"
```
configuration seems wrong for what takes a constant

Reply:
According to service team, these are not constants.